pingcap/tiup
# Extract get audit list

### What problem does this PR solve? 

Similar to PR #1139 , this PR is used to support PR #1172 (tiup web UI), so we can show the audit list in the web UI without parsing the command line output.

![image](https://user-images.githubusercontent.com/1284531/111329282-f8ff2d80-86a9-11eb-81d2-44c54e6b2a1b.png)


### What is changed and how it works?

Extract a GetAuditList() method

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch

Release notes:

```release-note
NONE
```

---
Maual test:

Output by this PR:

```
> bin/tiup-cluster audit
ID           Time                       Command
--           ----                       -------
fzgtCbNvFgm  2021-03-16T22:32:18+08:00  bin/tiup-cluster check multi-host --cluster
fzgsZxPGbXs  2021-03-16T22:23:00+08:00  bin/tiup-cluster check multi-host --cluster
fzgsWNLWbg1  2021-03-16T22:22:12+08:00  /Users/baurine/.tiup/components/cluster/v1.3.5/tiup-cluster check multi-host --cluster
fzgsKrtgpBY  2021-03-16T22:19:10+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster check multi-host --cluster
fzgsz4QJrcb  2021-03-16T22:16:43+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster display multi-host
fzgsyNsvtGQ  2021-03-16T22:16:38+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster list
fzg6gBm308p  2021-03-16T17:58:12+08:00  bin/tiup-cluster check multi-host --cluster
...
```

Output by master branch code:

```
> bin/tiup-cluster audit
ID           Time                       Command
--           ----                       -------
fzgvJqdqkPd  2021-03-16T22:48:47+08:00  bin/tiup-cluster audit
fzgtCbNvFgm  2021-03-16T22:32:18+08:00  bin/tiup-cluster check multi-host --cluster
fzgsZxPGbXs  2021-03-16T22:23:00+08:00  bin/tiup-cluster check multi-host --cluster
fzgsWNLWbg1  2021-03-16T22:22:12+08:00  /Users/baurine/.tiup/components/cluster/v1.3.5/tiup-cluster check multi-host --cluster
fzgsKrtgpBY  2021-03-16T22:19:10+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster check multi-host --cluster
fzgsz4QJrcb  2021-03-16T22:16:43+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster display multi-host
fzgsyNsvtGQ  2021-03-16T22:16:38+08:00  /Users/baurine/.tiup/components/cluster/v1.3.4/tiup-cluster list
fzg6gBm308p  2021-03-16T17:58:12+08:00  bin/tiup-cluster check multi-host --cluster
...
```
